lib/ansible/cli/doc.py

```diff
@@ -829,8 +839,8 @@ def format_plugin_doc(plugin, plugin_type, doc, plainexamples, returndocs, metad
         doc['returndocs'] = returndocs
         doc['metadata'] = metadata
 
-        if context.CLIARGS['show_snippet'] and plugin_type == 'module':
-            text = DocCLI.get_snippet_text(doc)
+        if context.CLIARGS['show_snippet']:
+            text = DocCLI.get_snippet_text(doc, plugin_type)
```

Comment:
Now that this no longer depends on plugin_type to activate the snippet path, this condition should be moved to the caller.  It's a decision about the overall format the data should be displayed as rather than a nitty-gritty decision about how to display a specific piece of data in this particular format. It's the roughly the same level of decision as whether the output should be formatted as a json string or as human readable output which is being done in the caller.

Reply:
It is something I have been thinking about for a while, making a greater separation between data sourcing and formatting, it will probably simplify things a lot, but that might be worth it's own PR.